### PR TITLE
README: clarify build system version

### LINKED ISSUE #120: Question about pyproject.toml syntax
The README for this project suggests using:
```toml
[build-system]
requires = ["jupyter_packaging~=0.10.0,<2"]
build-backend = "setuptools.build_meta"
```
However, doesn't `~=0.10.0` imply `>=0.10.0,==0.10.*`? So not only is the `<2` unnecessary, it's also misleading since it suggests that version 0.11.0 or 1.X would work? I'm referring to [this syntax specification](https://www.python.org/dev/peps/pep-0440/#compatible-release).

### PR DESCRIPTION
**Pull request by adamjstewart**
Closes #120

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ An example:
 
 ```toml
 [build-system]
-requires = ["jupyter_packaging~=0.10.0,<2"]
+requires = ["jupyter_packaging~=0.10.0"]
```
Comment (blink1073) [inline README.md:21]: Sorry, I should have specified, let's keep compatibility with 1.x+ and use `~=0.10,<2`

Comment (adamjstewart) [reply README.md:21]: `~=0.10,<2` implies `>=0.10,==0.X`, it doesn't include 1.X. Do you instead want to use `>=0.10,<2`?

Comment (blink1073) [reply README.md:21]: Yes, please, the tilde spec is quite confusing.  :smile:

**Review by blink1073 — APPROVED** [review]
Thank you!

**Pull request merged**